iroha/src/sumeragi.rs

```diff
@@ -45,8 +51,65 @@ impl Message {
     }
 }
 
+/// Structure describing a request to other peers to change view because of the commit timeout.
+#[derive(Io, Decode, Encode, Debug, Clone)]
+pub struct CommitTimeout {
+    voting_block_hash: Hash,
+    signatures: Vec<Signature>,
+}
+
+impl CommitTimeout {
+    /// Initializes a new commit timeout message with no signatures
+    pub fn new(voting_block: VotingBlock) -> CommitTimeout {
+        CommitTimeout {
+            voting_block_hash: voting_block.block.hash(),
+            signatures: vec![],
+        }
+    }
+
+    /// Signes this request with the peer's public and private key.
+    /// This way peers vote for changing the view.
+    pub fn sign(&mut self, public_key: &PublicKey, private_key: &PrivateKey) -> Result<(), String> {
+        let already_signed = self
+            .signatures
+            .iter()
+            .any(|signature| signature.public_key() == public_key);
+        if !already_signed {
+            self.signatures.push(Signature::new(
+                *public_key,
+                &self.voting_block_hash,
+                private_key,
+            )?);
+            Ok(())
+        } else {
+            Err("Already signed.".to_string())
+        }
+    }
+}
+
+/// Structure represents a block that is currently in discussion.
+#[derive(Debug, Clone)]
+pub struct VotingBlock {
+    /// At what time has this peer voted for this block
+    voted_at: Duration,
+    /// Signed Block hash
+    block: SignedBlock,
+}
+
+impl VotingBlock {
+    /// Initializes new block in discussion and assigns current time to `voted_at` field.
```

Comment:
We can't provide information about usages of this `pub` method. It may be initialized anywhere.
Also - better to use term "Constructs" if we have no 'init' like logic inside.

Reply:
So is it better to make a separate method `vote` and call it after new?

Reply:
No - just rewrite the doc